### ISSUE TITLE
CodeCov

### ISSUE THREAD

**Issue by lrouhier** [ivadomed/ivadomed#343]
The codecov on the main Readme.md is not working. This is an expected metric for JMLR submission. It needs to be fixed.

**Comment by lrouhier** [comment]
[report.txt](https://github.com/ivadomed/ivadomed/files/4914631/report.txt)
 Report of code coverage so far (done locally trying to figure out a way of having an automatic report)

**Comment by lrouhier** [comment]
We can have an automatic report here https://coveralls.io/github/ivadomed/ivadomed?branch=lr/increase_cov, however there is a few issue, training code will be a pain to run (as well as evaluation and other). We have something that works, yet, we are not close at all to 100 %. 

I went to check Monai and they basically don't have the training loop as a script, but they show it onside a tutorial so they don't need to test it (and achieve 94 % coverage).

**Comment by jcohenadad** [comment]
> I went to check Monai and they basically don't have the training loop as a script, but they show it onside a tutorial so they don't need to test it (and achieve 94 % coverage).

is code coverage affected by scripts _outside_ the ivadomed package? if so maybe we can consider moving them out?

**Comment by lrouhier** [comment]
We can choose the base folder to be ivadomed and in that case we won't consider the one _outside_.  We can also exclude files (as explained in pytest-cov [doc](https://pytest-cov.readthedocs.io/en/latest/config.html)). 
We can think about it because then we'll have a good score but some uncovered score, I don't know how that would be perceive

**Comment by jcohenadad** [comment]
> We can think about it because then we'll have a good score but some uncovered score, I don't know how that would be perceive

if there is a good justification to not cover part of the code, it should be fine. and it seems like this situation (training script) is a pretty good justification. @Drulex ?

**Comment by Drulex** [comment]
I think it's a matter of preference. Personally I'm not too impressed by the coverage % metric which focuses on lines/path covered (but doesn't consider the sequences/logical paths).

I would say that everything in the core api `ivadomed.ivadomed` needs to be covered. In general, every feature of the package should be covered. If the package offers training functionality, then it would make sense to cover that functionality, unless I am misunderstanding something?

**Comment by lrouhier** [comment]
This was closed thanks to multiple PR (#422 , #414, #385, #406 , #403, #408, #410 , #412 ). Code cov is now at 91 %
